snap/info_snap_yaml.go

```diff
@@ -508,13 +493,58 @@ func bindUnboundSlots(slotNames []string, snap *Info) error {
 				slot.Apps[appName] = app
 			}
 			for hookName, hook := range snap.Hooks {
+				if hook.Slots == nil {
```

Comment:
the code in bindUnboundPlugs doesn't do this, is it buggy? are we missing tests?

Reply:
Thanks for spotting, that was actually precautionary and yes, I forgot to add it here as well. It's not possible to hit this cases because `setHooksFromSnapYaml` and `setAppsFromSnapYaml' initialize these maps.

Reply:
Some corner case of only implicit hooks might trigger that I think anyway, as I said still feeling like we might be missing some tests

Reply:
Added two more test cases, it's however not possible to hit this case, this is based on the fact that 1) `setHooksFromSnapYaml` initializes `hook.Slots` and `hook.Plugs` for all explicitely defined hooks and 2) `bindUnboundSlots` and `bindUnboundPlugs` is only run for explicitely defined hooks. Implicit hooks are added later and considered by the new method I added.

Reply:
Yes. I just went ahead and removed it.